### DISCUSSION TITLE
Day 5 Level 1 Q14

### DISCUSSION THREAD

**Discussion by Harufuu** [Asabeneh/30-Days-Of-JavaScript#772] [Q&A]
```js
let itCompanies = [
    "Facebook",
    "Google",
    "Microsoft",
    "Apple",
    "IBM",
    "Oracle",
    "Amazon"
  ]
// Filter out companies which have more than one 'o' without the filter method
```

I solve it with the For loop, something like this
```js
let filtered_array = [];
for(i = 0; i < itCompanies.length; i++) {
  let thing = itCompanies[i];
  if(thing.indexOf("o") !== thing.lastIndexOf("o")){
    filtered_array.push(thing);
  }
}

console.log(filtered_array) // [ 'Facebook', 'Google', 'Microsoft' ] 
```

The problem is that the loop stuff would first appear in the course on Day 6, while this is a question in the Day 5 exercise. So, I wonder if there is a solution that does not involve either the filter method or the loop.

**Comment by davinicikrypto** [comment]
So all we just have to do is when you a question and it so confusing just
leave it blank and move on with the next one then come back to that
question again when you have more confidence.

**Comment by ghost** [comment]
@Harufuu Here, it may help you.

```
let itCompanies = ['facebook', 'google', 'microsoft',  .....]
var newArray = []
itCompanies.map((v,i)=>{
if(v.includes('o')){
newArray.push(v);
}
})
console.log(newArray); // [items with o will be added here.]
```

**Comment by Jouqio** [comment]
Your for loop solution is correct  `indexOf('o') !== lastIndexOf('o')` is actually 
a smart trick to detect more than one occurrence, and your output is right.

To answer the core question: there's no practical way to filter an array without 
some form of iteration. Even `.filter()`, `.map()`, and `.reduce()` are just loops 
under the hood. The hint "without the filter method" means "use a different approach", 
not "avoid all loops entirely". Your for loop is exactly what's expected here.

Worth noting: the previous `.map()` suggestion in this thread doesn't actually solve 
the problem correctly. It checks `.includes('o')` which matches *any* 'o', not 
*more than one*. With that approach, "Amazon" would be incorrectly included.

One alternative way to express the condition (though you still need a loop to apply it):

```js
company.split('o').length > 2  // true when string has 2+ 'o' characters
```

But ultimately, iterating is unavoidable. Your for loop is the right approach for 
Day 5  the exercise says "without the filter method", not "without loops". 
Using a for loop here is the intended solution.
